dbt_server/state.py

```diff
@@ -14,18 +14,40 @@
 
 @dataclass
 class CachedManifest:
+    """CachedManifest holds a local in memory cache of latest manifest and
+    other project metadata.
+
+    This class is singleton.
+    This class is thread-compatible as most write operations are guarded by
+    lock."""
+
+    # State id is a string generated by client of dbt-server, dbt-server takes
+    # it as a manifest version. It's optional and can be None.
     state_id: Optional[str] = None
+    # Indicates local dbt project path. It might be None when clients push
+    # source code to dbt-server thus we don't have a local project path.
```

Comment:
```suggestion
    # Indicates local dbt project path and is provided on parse. It might be None when
    # clients push source code to dbt-server and thus provide a state_id instead. 
    # Either state_id or project_path must be provided.
```